### PR TITLE
fix: make dict, ternary FunCall, and SVG path formatting idempotent

### PR DESCRIPTION
**Pull request by jucallej**
## Problem

Running `prettier --write` twice on the same HubL template should produce the same output both times (`format(format(x)) === format(x)`). Several patterns violated that, so `prettier --check` kept reporting changes even when the file was already formatted.

| Area | Symptom | Typical source |
|------|---------|----------------|
| **Dict literals** | Pass 1 adds/removes trailing commas or changes line breaks; pass 2 reverses it | `{% set lang = { "en": "...", ... , } %}` |
| **FunCall + ternary** | Argument line-breaking flips between single-line and multiline when a ternary is an argument | `renderCTA(cta, loop.index == 1 ? "primary" : "secondary", ...)` |
| **Inline SVG / path `d`** | HTML Prettier reflows multiline `d="..."` differently on each preprocess pass | Long `<path d="M... C...">` inside templates/macros |

The plugin formats HubL in two stages:

1. **Preprocess** — tokenize HubL, run stock HTML Prettier, restore tokens  
2. **Print** — parse HubL AST and print with `printHubl.ts`

Dict/ternary oscillation came from **stage 2** (unstable printer choices). SVG oscillation came from **stage 1** (HTML formatter touching path data, then the HubL printer re-indenting preserved content inconsistently).

---

### HubL printer — stable output for dicts and ternary calls

- **`Dict`** — always print one key per line, no trailing comma on the last entry (matches parser behavior).
- **`FunCall`** — when any argument is a `Ternary`, use a fixed multiline argument layout (`indent` + `softline` / `line`) so line breaks do not flip between runs.

### Preprocess + printer — keep SVG verbatim but indentable

1. **`preserveSvgElements`** (in `tokenize`) — replace each `<svg>...</svg>` with a placeholder **before** HTML Prettier runs, so path `d` data is never reflowed.
2. **`unTokenize`** — restore real SVG markup.
3. **`wrapSvgWithPreserve`** — wrap restored SVG in `{% preserve %}...{% endpreserve %}` so the HubL parser emits a `Preserve` node (same mechanism as `<pre>`).
4. **`printSvgPreserveContent`** — print preserved SVG with `join(hardline, lines)` and normalized relative indent so macro/block `indent()` applies consistently (without re-running HTML Prettier on the SVG).

```mermaid
flowchart TD
  T[trim + tokenize] --> T1[HubL / style / script placeholders]
  T1 --> T2[preserveSvgElements → svgblock comment]
  T2 --> H[HTML Prettier]
  H --> U[unTokenize — restore SVG + HubL]
  U --> W[wrapSvgWithPreserve]
  W --> P[preserveFormatting for pre]
  P --> PARSE[HubL parse + printHubl]
```

Order matters: SVG must be restored from placeholders **before** `wrapSvgWithPreserve` can match `<svg>`, and preserve wrappers are applied **after** HTML format so HubL tags are not fed through the HTML parser.

### CODE REVIEW

File: prettier/src/index.ts [hunk 1]
```diff
@@ -27,9 +27,43 @@ const Token = {
   npe: (index: number) => `npe${index}_`,
   comment: (index: number) => `<!--${index}-->`,
   placeholder: (index: number) => `<!--placeholder-${index}-->`,
+  svgBlock: (index: number) => `<!--svgblock-${index}-->`,
   jsonBlock: (match: string) => `{% json_block %}${match}{% end_json_block %}`,
 };
 
+const SVG_ELEMENT_REGEX = /<svg\b[\s\S]*?<\/svg>/gim;
```
Comment (TanyaScales) [inline prettier/src/index.ts:40]: This doesn't account for nested SVGs -- proooooobably not a common thing/edge case. But just a note in case it comes up in the future.

File: prettier/tests/__snapshots__/run_tests.js.snap [hunk 2]
```diff
@@ -1187,6 +1187,67 @@ screenshotPath: ../images/template-previews/hubdb.png
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+{% macro renderMobileBackground() %}
+  <div class="insights-layout-background-container -mobile">
+    <svg width="375" height="285" viewBox="0 0 375 285" fill="none" xmlns="http://www.w3.org/2000/svg">
+    <path
+      d="M543.279 662.304C536.98 666.635 529.524 668.971 521.846 669.878
+        C490.947 673.315 460.01 676.64 429.035 680.228 333.115 691.354
+        237.195 702.48 141.275 713.606 45.355 724.732"
+    />
+  </svg>
```
Comment (TanyaScales) [inline prettier/tests/__snapshots__/run_tests.js.snap]: The indentation here in the snapshot doesn't match what I think is being expressed above: 

"the opening tag sits at the block indent and every other line keeps its offset relative to it. Because normalization is purely relative, re-formatting the output reproduces it exactly"

So, the SVG is removed from the formatting and preserved and the inserted without any changes, but this part of the snapshot has the indentation slightly off from that.

**Review by TanyaScales — DISMISSED** [review]
Approving with a couple comments - Feel free to merge after you take a look at the one re: the snapshot

File: prettier/src/index.ts (hunk 1 above)
Comment (jucallej) [reply prettier/src/index.ts:40]: Great point! I've added a comment to mention that nested svgs are not suported yet so it's easier to notice if it comes up in the future!

File: prettier/tests/__snapshots__/run_tests.js.snap (hunk 2 above)
Comment (jucallej) [reply prettier/tests/__snapshots__/run_tests.js.snap]: Great point, you are right! I've made some changes in the last commits, and it the indentation should be preserved. Let me know if I'm missing anything else!
